### ISSUE TITLE
Main & Zone2 support is close but not yet working as expected

### ISSUE THREAD

**Issue by ryanschmidt** [ToddGreenfield/homebridge-onkyo#73]
- Homebridge version: 0.4.53
- Plugin version: 0.7.16
- OS, including release name/version: Ubuntu, 18.04
- [ x ] Homebridge-Onkyo installed via OS/distribution mechanisms
- [ x ] You enabled homebridge debug output

**Describe the bug**
I believe the Zone2 functionality works but it breaks the Main when it's enabled. It appears to work when you just enable the receiver inputs using the Home app but when you turn off the Receiver Main from the physical receiver while Zone2 config is in place, the Main never turns off. Zone2 works fine when you have both and Main works fine when you only have the Main config in place.

**To Reproduce**
Information on your Onkyo receiver:
1. Model information
I have an Integra DRC-R1. It works as intended using TX-NR609 except for the Zone2 functionality.

Information on your configuration:
1. Homebridge config

```
        {
            "platform": "Onkyo",
            "receivers": [
                {
                    "model": "TX-NR609",
                    "ip_address": "192.168.85.157",
                    "poll_status_interval": "900",
                    "name": "Receiver",
                    "zone": "main",
                    "default_input": "net",
                    "default_volume": "10",
                    "max_volume": "40",
                    "inputs": [
                        {
                            "input_name": "TV",
                            "display_name": "TV"
                        },
                        {
                            "input_name": "AUX",
                            "display_name": "PS4"
                        }
                    ],
                    "map_volume_100": false,
                    "volume_dimmer": false,
                    "switch_service": false,
                    "filter_inputs": true
                },
                {
                    "model": "TX-NR609",
                    "ip_address": "192.168.85.157",
                    "poll_status_interval": "900",
                    "name": "Receiver Zone 2",
                    "zone": "zone2",
                    "default_input": "net",
                    "default_volume": "10",
                    "max_volume": "40",
                    "inputs": [
                        {
                            "input_name": "TV",
                            "display_name": "TV"
                        },
                        {
                            "input_name": "AUX",
                            "display_name": "PS4"
                        }
                    ],
                    "map_volume_100": false,
                    "volume_dimmer": false,
                    "switch_service": false,
                    "filter_inputs": true
                }
            ]
        },
```

Information from the log:
1. Homebridge log file showing relevant information. Experienced users are asked to turn debug on.

Even with the below errors, Homebridge doesn't crash. It acts like it will work, both Receiver and Receiver Zone 2 show in Home and turning them on or off from within the app appear to work. What's broke is that they don't seem to get a response back from the physical receiver because of the shared use of the IP address.

```
[3/28/2020, 8:58:14 PM] [Onkyo] eventError: ERROR (server_error) Server error on 192.168.85.157:60128 - Error: connect EALREADY 192.168.85.157:60128 - Local (192.168.85.33:43908)
    at internalConnect (net.js:908:16)
    at defaultTriggerAsyncIdScope (internal/async_hooks.js:301:12)
    at net.js:999:9
    at processTicksAndRejections (internal/process/task_queues.js:75:11) {
  errno: 'EALREADY',
  code: 'EALREADY',
  syscall: 'connect',
  address: '192.168.85.157',
  port: 60128
}
[3/28/2020, 8:58:14 PM] [Onkyo] eventError: ERROR (server_error) Server error on 192.168.85.157:60128 - Error: connect EALREADY 192.168.85.157:60128 - Local (192.168.85.33:43908)
    at internalConnect (net.js:908:16)
    at defaultTriggerAsyncIdScope (internal/async_hooks.js:301:12)
    at net.js:999:9
    at processTicksAndRejections (internal/process/task_queues.js:75:11) {
  errno: 'EALREADY',
  code: 'EALREADY',
  syscall: 'connect',
  address: '192.168.85.157',
  port: 60128
}
```

```
[3/28/2020, 8:58:14 PM] [Onkyo] eventError: ERROR (send_not_connected) Not connected, can't send data: "PWRQSTN"
[3/28/2020, 8:58:14 PM] [Onkyo] eventError: ERROR (send_not_connected) Not connected, can't send data: "PWRQSTN"
[3/28/2020, 8:58:14 PM] [Onkyo] getPowerState - PWR QRY: ERROR - current state: false
[3/28/2020, 8:58:14 PM] [Onkyo] eventError: ERROR (send_not_connected) Not connected, can't send data: "SLIQSTN"
[3/28/2020, 8:58:14 PM] [Onkyo] eventError: ERROR (send_not_connected) Not connected, can't send data: "SLIQSTN"
[3/28/2020, 8:58:14 PM] [Onkyo] getInputState - INPUT QRY: ERROR - current i_state: 1
[3/28/2020, 8:58:14 PM] [Onkyo] eventError: ERROR (send_not_connected) Not connected, can't send data: "AMTQSTN"
[3/28/2020, 8:58:14 PM] [Onkyo] eventError: ERROR (send_not_connected) Not connected, can't send data: "AMTQSTN"
[3/28/2020, 8:58:14 PM] [Onkyo] getMuteState - MUTE QRY: ERROR - current m_state: false
[3/28/2020, 8:58:14 PM] [Onkyo] eventError: ERROR (send_not_connected) Not connected, can't send data: "MVLQSTN"
[3/28/2020, 8:58:14 PM] [Onkyo] eventError: ERROR (send_not_connected) Not connected, can't send data: "MVLQSTN"
[3/28/2020, 8:58:14 PM] [Onkyo] getVolumeState - VOLUME QRY: ERROR - current v_state: 0
[3/28/2020, 8:58:14 PM] [Onkyo] eventError: ERROR (send_not_connected) Not connected, can't send data: "ZPWQSTN"
[3/28/2020, 8:58:14 PM] [Onkyo] eventError: ERROR (send_not_connected) Not connected, can't send data: "ZPWQSTN"
[3/28/2020, 8:58:14 PM] [Onkyo] getPowerState - PWR QRY: ERROR - current state: false
[3/28/2020, 8:58:14 PM] [Onkyo] eventError: ERROR (send_not_connected) Not connected, can't send data: "SLZQSTN"
[3/28/2020, 8:58:14 PM] [Onkyo] eventError: ERROR (send_not_connected) Not connected, can't send data: "SLZQSTN"
[3/28/2020, 8:58:14 PM] [Onkyo] getInputState - INPUT QRY: ERROR - current i_state: 1
[3/28/2020, 8:58:14 PM] [Onkyo] eventError: ERROR (send_not_connected) Not connected, can't send data: "ZMTQSTN"
[3/28/2020, 8:58:14 PM] [Onkyo] eventError: ERROR (send_not_connected) Not connected, can't send data: "ZMTQSTN"
[3/28/2020, 8:58:14 PM] [Onkyo] getMuteState - MUTE QRY: ERROR - current m_state: false
[3/28/2020, 8:58:14 PM] [Onkyo] eventError: ERROR (send_not_connected) Not connected, can't send data: "ZVLQSTN"
[3/28/2020, 8:58:14 PM] [Onkyo] eventError: ERROR (send_not_connected) Not connected, can't send data: "ZVLQSTN"
[3/28/2020, 8:58:14 PM] [Onkyo] getVolumeState - VOLUME QRY: ERROR - current v_state: 0
```

**Expected behavior**
When enabling Main and Zone2 config, I would except that when turning Main or Zone2 receiver on or off from the physical receiver, the Home switch for both would also turn on or off.

**Comment by solowalker27** [comment]
I think this might be due to the receiver only accepting one connection at a time. The zone2 being a second device in HomeKit might be trying to use a second connection to the receiver which would cause this error. If this is the case, I'm not entirely sure how to solve it. Might be able to try having zone2 be an option in the config and then add it as a linked service maybe? That might work if zone2 is just another On/Off switch. Is that how it works or would that be acceptable?

I personally don't use zone2. I might be able to try to just play with it. Do you control volume and source for zone2 separately or does it the same as main?

**Comment by ryanschmidt** [comment]
Source can be the same or different for zone2. Thanks for your help!

**Comment by toshibochan** [comment]
I wonder if do new homebridge-onkyo-zone2 will work?

**Comment by solowalker27** [comment]
Not if the issue is only allowing a single connection to the unit at a time. A second plugin would be trying to make a second connection. It'd have to be handled by the same copy of the plugin that's doing the main control but would be another accessory.

**Comment by jondthompson** [comment]
If the issue is because of the same IP address (and it probably is), it should be resolved in the 0.8.1+

**Comment by ryanschmidt** [comment]
Confirmed this is working as intended in 0.8.2. Thanks!
